optuna/storages/_rdb/alembic/versions/v3.0.0.c.py

```diff
@@ -89,20 +89,30 @@ def upgrade():
 
     session = orm.Session(bind=bind)
     try:
-        records = session.query(IntermediateValueModel).all()
+        records = (
+            session.query(IntermediateValueModel)
+            .filter(
+                sa.or_(
+                    IntermediateValueModel.intermediate_value > 1e16,
+                    IntermediateValueModel.intermediate_value < -1e16,
+                    IntermediateValueModel.intermediate_value.is_(None),
```

Comment:
I confirmed that `nan` values are stored as NULL on `SQLite3`, `MySQL` and `PostgreSQL` with [optuna-e2e](https://github.com/c-bata/optuna-e2e/tree/optuna-303-data-migration).

<details>
<summary>SQLite3</summary>

```
$ sqlite3 ./data/sample.db
sqlite> .mode column
sqlite> .header on
sqlite> select * from trial_intermediate_values;
trial_intermediate_value_id  trial_id  step  intermediate_value
---------------------------  --------  ----  ---------------------
1                            1         1     3.40282346638529e+38
2                            1         2     -3.40282346638529e+38
3                            1         3
4                            2         1     3.40282346638529e+38
5                            2         2     -3.40282346638529e+38
6                            2         3
7                            3         1     3.40282346638529e+38
8                            3         2     -3.40282346638529e+38
9                            3         3
10                           4         1     3.40282346638529e+38
11                           4         2     -3.40282346638529e+38
12                           4         3
13                           5         1     3.40282346638529e+38
14                           5         2     -3.40282346638529e+38
15                           5         3
16                           6         1     3.40282346638529e+38
17                           6         2     -3.40282346638529e+38
18                           6         3
19                           7         1     3.40282346638529e+38
20                           7         2     -3.40282346638529e+38
21                           7         3
22                           8         1     3.40282346638529e+38
23                           8         2     -3.40282346638529e+38
24                           8         3
25                           9         1     3.40282346638529e+38
26                           9         2     -3.40282346638529e+38
27                           9         3
28                           10        1     3.40282346638529e+38
29                           10        2     -3.40282346638529e+38
30                           10        3

sqlite> select * from trial_intermediate_values where intermediate_value is null;
trial_intermediate_value_id  trial_id  step  intermediate_value
---------------------------  --------  ----  ------------------
3                            1         3
6                            2         3
9                            3         3
12                           4         3
15                           5         3
18                           6         3
21                           7         3
24                           8         3
27                           9         3
30                           10        3
```


</details>

<details>
<summary>MySQL</summary>

```
$ mysql --host 127.0.0.1 --port 3306 --user root -D optuna -proot

mysql> select * from trial_intermediate_values;
+-----------------------------+----------+------+--------------------+
| trial_intermediate_value_id | trial_id | step | intermediate_value |
+-----------------------------+----------+------+--------------------+
|                           1 |        1 |    1 |         3.40282e38 |
|                           2 |        1 |    2 |        -3.40282e38 |
|                           3 |        1 |    3 |               NULL |
|                           4 |        2 |    1 |         3.40282e38 |
|                           5 |        2 |    2 |        -3.40282e38 |
|                           6 |        2 |    3 |               NULL |
|                           7 |        3 |    1 |         3.40282e38 |
|                           8 |        3 |    2 |        -3.40282e38 |
|                           9 |        3 |    3 |               NULL |
|                          10 |        4 |    1 |         3.40282e38 |
|                          11 |        4 |    2 |        -3.40282e38 |
|                          12 |        4 |    3 |               NULL |
|                          13 |        5 |    1 |         3.40282e38 |
|                          14 |        5 |    2 |        -3.40282e38 |
|                          15 |        5 |    3 |               NULL |
|                          16 |        6 |    1 |         3.40282e38 |
|                          17 |        6 |    2 |        -3.40282e38 |
|                          18 |        6 |    3 |               NULL |
|                          19 |        7 |    1 |         3.40282e38 |
|                          20 |        7 |    2 |        -3.40282e38 |
|                          21 |        7 |    3 |               NULL |
|                          22 |        8 |    1 |         3.40282e38 |
|                          23 |        8 |    2 |        -3.40282e38 |
|                          24 |        8 |    3 |               NULL |
|                          25 |        9 |    1 |         3.40282e38 |
|                          26 |        9 |    2 |        -3.40282e38 |
|                          27 |        9 |    3 |               NULL |
|                          28 |       10 |    1 |         3.40282e38 |
|                          29 |       10 |    2 |        -3.40282e38 |
|                          30 |       10 |    3 |               NULL |
+-----------------------------+----------+------+--------------------+
30 rows in set (0.01 sec)
```

</details>

<details>
<summary>PostgreSQL</summary>

```
$ docker exec -it optuna-e2e-postgresql-1 /bin/bash
root@c7a20f532829:/# psql optuna
psql (12.10 (Debian 12.10-1.pgdg110+1))
Type "help" for help.

optuna=# select * from trial_intermediate_values;
 trial_intermediate_value_id | trial_id | step |   intermediate_value
-----------------------------+----------+------+-------------------------
                           1 |        1 |    1 |  3.4028234663852886e+38
                           2 |        1 |    2 | -3.4028234663852886e+38
                           3 |        1 |    3 |
                           4 |        2 |    1 |  3.4028234663852886e+38
                           5 |        2 |    2 | -3.4028234663852886e+38
                           6 |        2 |    3 |
                           7 |        3 |    1 |  3.4028234663852886e+38
                           8 |        3 |    2 | -3.4028234663852886e+38
                           9 |        3 |    3 |
                          10 |        4 |    1 |  3.4028234663852886e+38
                          11 |        4 |    2 | -3.4028234663852886e+38
                          12 |        4 |    3 |
                          13 |        5 |    1 |  3.4028234663852886e+38
                          14 |        5 |    2 | -3.4028234663852886e+38
                          15 |        5 |    3 |
                          16 |        6 |    1 |  3.4028234663852886e+38
                          17 |        6 |    2 | -3.4028234663852886e+38
                          18 |        6 |    3 |
                          19 |        7 |    1 |  3.4028234663852886e+38
                          20 |        7 |    2 | -3.4028234663852886e+38
                          21 |        7 |    3 |
                          22 |        8 |    1 |  3.4028234663852886e+38
                          23 |        8 |    2 | -3.4028234663852886e+38
                          24 |        8 |    3 |
                          25 |        9 |    1 |  3.4028234663852886e+38
                          26 |        9 |    2 | -3.4028234663852886e+38
                          27 |        9 |    3 |
                          28 |       10 |    1 |  3.4028234663852886e+38
                          29 |       10 |    2 | -3.4028234663852886e+38
                          30 |       10 |    3 |
(30 rows)

optuna=# select * from trial_intermediate_values where intermediate_value is NULL;
 trial_intermediate_value_id | trial_id | step | intermediate_value
-----------------------------+----------+------+--------------------
                           3 |        1 |    3 |
                           6 |        2 |    3 |
                           9 |        3 |    3 |
                          12 |        4 |    3 |
                          15 |        5 |    3 |
                          18 |        6 |    3 |
                          21 |        7 |    3 |
                          24 |        8 |    3 |
                          27 |        9 |    3 |
                          30 |       10 |    3 |
(10 rows)
```

</details>

Reply:
An already-converted `inf` values are also represented as `None`, so this would make the migration non-idempotent.